### PR TITLE
[feature](planner) Support `select constant from dual` syntax sugar

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

In MySQL, it's common to use a simplified syntax like `SELECT constant FROM dual` which is equivalent to just `SELECT constant`. This syntax is often used by BI tools when utilizing MySQL connectors to verify connection validity. To enhance compatibility and ensure seamless integration with such tools, we have now implemented this feature in Doris.

### Key Changes:
- Doris now interprets `SELECT constant FROM dual` as `SELECT constant`, aligning with MySQL's behavior.
- This update ensures that BI tools can use standard MySQL connectors without modifications or errors when connecting to Doris.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by zy-kkk** [comment]
run buildall

**Comment by zclllyybb** [comment]
what's this grammar for?

**Comment by morrySnow** [comment]
add feature desc please

File: regression-test/suites/query_p0/dual/dual.groovy [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+suite('dual') {
+
+    qt_sql 'select 1 from dual'
+    qt_sql 'select 1 from dual where 1'
+    qt_sql 'select 1 from dual where 1 = 1'
+    qt_sql 'select 1 from dual where 0'
```
Comment (morrySnow) [inline regression-test/suites/query_p0/dual/dual.groovy:23]: please add more complex test case. such as in cte, in subquery, in set operand, use with group by, window and other syntax

**Comment by zy-kkk** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**